src/connectors/impls/gcl/writer.rs

```diff
@@ -0,0 +1,239 @@
+    // "billingAccounts/\[BILLING_ACCOUNT_ID]/logs/[LOG_ID\]"
+    // "folders/\[FOLDER_ID]/logs/[LOG_ID\]"
+    // \[LOG_ID\] must be URL-encoded. For example:
+    //
+    // "projects/my-project-id/logs/syslog"
+    // "organizations/1234567890/logs/cloudresourcemanager.googleapis.com%2Factivity"
+    // The permission logging.logEntries.create is needed on each project, organization, billing account, or folder that is receiving new log entries, whether the resource is specified in logName or in an individual log entry.
+    pub log_name: Option<String>,
+
+    // Optional. A default monitored resource object that is assigned to all log entries in entries that do not specify a value for resource. Example:
+    //
+    // { "type": "gce_instance",
+    // "labels": {
+    //   "zone": "us-central1-a", "instance_id": "00000000000000000000" }}
+    #[serde(default = "default_resource")]
+    pub resource: Option<simd_json::OwnedValue>,
+
+    // Optional. Whether valid entries should be written even if some other entries fail due to INVALID_ARGUMENT or PERMISSION_DENIED errors. If any entry is not written, then the response status is the error associated with one of the failed entries and the response includes error details keyed by the entries' zero-based index in the entries.write method.
+    #[serde(default = "default_partial_success")]
+    pub partial_success: bool,
+
+    // Optional. If true, the request should expect normal response, but the entries won't be persisted nor exported. Useful for checking whether the logging API endpoints are working properly before sending valuable data.
+    #[serde(default = "default_dry_run")]
+    pub dry_run: bool,
+
+    // pub table_id: String,
+    #[serde(default = "default_connect_timeout")]
+    pub connect_timeout: u64,
+
+    // Default Log severity
+    #[serde(default = "default_log_severity")]
+    pub default_severity: Option<i32>,
+
+    #[serde(default = "default_labels")]
+    pub labels: Option<HashMap<String, String>>,
+}
+
+fn default_resource() -> Option<simd_json::OwnedValue> {
+    None
+}
+
+fn default_partial_success() -> bool {
+    false
+}
+
+fn default_dry_run() -> bool {
+    false
+}
+
+fn default_connect_timeout() -> u64 {
+    0
+}
+
+#[allow(clippy::unnecessary_wraps)] // Allow for deserialization from value
+fn default_log_severity() -> Option<i32> {
+    Some(LogSeverity::Info as i32)
+}
+
+fn default_labels() -> Option<HashMap<String, String>> {
+    None
+}
+
+impl ConfigImpl for Config {}
+
+fn value_to_monitored_resource(
+    from: Option<&simd_json::OwnedValue>,
+) -> Result<Option<MonitoredResource>> {
+    match from {
+        None => Ok(None),
+        Some(from) => {
+            let vt = from.value_type();
+            match from {
+                OwnedValue::Object(from) => {
+                    let kind = from.get("type");
+                    let kind = kind.as_str();
+                    let maybe_labels = from.get("labels");
```

Comment:
`unwrap_or_default` would save the match and nesting below

Reply:
Didn't attend to this one, not seeing how unwrap_or_default improves readability here

Reply:
It would allow collapsing the next statment down to:
```rust
                    let labels: HashMap<_, _> = from.try_get_object("labels")?
                            .cloned()
                            .unwrap_or_default()
                            .iter()
                            .map(|(key, value)| {
                                let key = key.to_string();
                                let value = value.to_string();
                                (key, value)
                            })
                            .collect();
```

Reply:
I just realiuze why this makes no sense :) in my mind #1819 is already merged and this code legal, right now it isn't so it can't seem useful yet

Reply:
Added #1819 to a comment to consider unknown_or_default in due course - closing for now